MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm

```diff
@@ -345,19 +354,81 @@ - (void)onProcessingLog:(id<MSLog>)log withPriority:(MSPriority)priority {
     NSData *serializedLog = [NSKeyedArchiver archivedDataWithRootObject:log];
     if (serializedLog && (serializedLog.length > 0)) {
 
-      // Our arrays contain a max of 20 items, so our maxIndex == 19.
-      int maxIndex = ms_crashes_log_buffer_size - 1;
-      if (self.bufferIndex[@(priority)].integerValue > maxIndex) {
+      NSNumber *oldestTimestamp;
+      NSNumberFormatter *timestampFormatter = [[NSNumberFormatter alloc] init];
+      timestampFormatter.numberStyle = NSNumberFormatterDecimalStyle;
+      long indexToDelete = 0;
+      for (auto it = msCrashesLogBuffer[priority].begin(), end = msCrashesLogBuffer[priority].end(); it != end; ++it) {
```

Comment:
Instead of iterating the buffer, you can have two indexes - one for the latest item (or new space to be written next time) and the other for oldest index. This way you don't need to find an empty space every time.